example/lib/src/ui/device_list.dart

```diff
@@ -53,17 +53,21 @@ class _DeviceListState extends State<_DeviceList> {
 
   bool _isValidUuidInput() {
     final uuidText = _uuidController.text;
-    try {
-      Uuid.parse(uuidText);
+    if (uuidText.isEmpty) {
       return true;
-    } on Exception {
-      return false;
+    } else {
+      try {
+        Uuid.parse(uuidText);
+        return true;
+      } on Exception {
+        return false;
+      }
     }
   }
 
   void _startScanning() {
-    final uuid = Uuid.parse(_uuidController.text);
-    widget.startScan(uuid);
+    final text = _uuidController.text;
+    widget.startScan(text.isEmpty ? null : [Uuid.parse(_uuidController.text)]);
```

Comment:
If you _are_ using a list here now, the why do you still use `null`?? Just pass an empty list.